### PR TITLE
Minimal fix for Algebraic Extensions over finite fields of order > 256.

### PR DESCRIPTION
**Pull request by stevelinton**
Code really needs rewriting, as does the whole issue of mechanisms to
construct/convert a vector in/to the most efficient form.

### CODE REVIEW

**Comment by stevelinton** [comment]
Addresses #1561

**Comment by stevelinton** [comment]
Very confused. The new test file runs fine when run using Test from the command-line, but fails when run as part of testbugfix.g or TestDirectory. Revised test file fixes various minor problems but does not seem to resolve this.

File: tst/testbugfix/2017-08-05-algfld.tst [hunk 1]
```diff
@@ -0,0 +1,14 @@
+#
+# Algebraic Extension used to fail for finite fields of size over 256
```
Comment (fingolfin) [inline tst/testbugfix/2017-08-05-algfld.tst]: Nitpick: "Algebraic Extension" -> "AlgebraicExtension"

File: tst/testbugfix/2017-08-05-algfld.tst [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+#
+gap> algexttest := function(q, i1,i2)
```
Comment (fingolfin) [inline tst/testbugfix/2017-08-05-algfld.tst]: Nitpcik: Either remove space after one comma, or add one after the other

File: tst/testbugfix/2017-08-05-algfld.tst [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+>    local f,x,pol;    
+>    f := GF(q);
+>    x :=Indeterminate(f,"x");
+>    pol := x^2+Z(q)^i1*x+Z(q)^i2;
+>    AlgebraicExtension(f,pol);
+> end;;
+gap> algexttest(1009,108,864); # from bug report
+gap> algexttest(1024,1023,5);  # prime under 256, field size over
+gap> algexttest(65537,1,1);   # prime over 2^16
```
Comment (fingolfin) [inline tst/testbugfix/2017-08-05-algfld.tst]: Nitpick: indention of this comment doesn't match the other comments

File: lib/algfld.gi [hunk 4]
```diff
@@ -105,7 +105,7 @@ local fam,i,cof,red,rchar,impattr,deg;
   fam!.deg:=deg;
   i:=List([1..DegreeOfLaurentPolynomial(p)],i->fam!.zeroCoefficient);
   i[2]:=fam!.oneCoefficient;
-  i:=ImmutableVector(rchar,i,true);
+  i:=ImmutableVector(Size(f),i,true);
```
Comment (fingolfin) [inline lib/algfld.gi]: This fix looks correct to me.

Nitpick on the commits: The message "Add test for this bug." is not so helpful. I'd either squash the two commits into one, or amend the commit message, say to "Add test for bug in AlgebraicExtension"

Comment (fingolfin) [inline lib/algfld.gi]: Also: If you add `Fixes #1561` or `Resolves #1561` to the commit message (say as its last line), then merging this PR will automatically close #1561.

**Comment by stevelinton** [comment]
I'll rebase and tidy once #1573 is merged

**Comment by fingolfin** [comment]
@stevelinton #1573 was merged, you can rebase

**Comment by stevelinton** [comment]
Rebased and addressed nitpicks. Ready to merge as far as I know.

**Pull request merged**